runtime/lua/vim/shared.lua

```diff
@@ -188,6 +188,31 @@ local function pesc(s)
   return s:gsub('[%(%)%.%%%+%-%*%?%[%]%^%$]', '%%%1')
 end
 
+--- Test if `prefix` is a prefix of `s`
+--
+-- @tparam string s String to check
+-- @tparam string prefix Potential prefix
+-- @treturn boolean True if prefix is a prefix of s
```

Comment:
The doc generator requires the `--` to be connected, like this (see existing code):

    --@foo

Reply:
I very much do! this would've been helpful for my own lua projects that I've been working on recently.

Reply:
> this would've been helpful for my own lua projects

Should be no problem, but it's somewhat hideous. We use the mentioned python script, plus doxygen, plus [this thing](https://github.com/neovim/neovim/blob/master/scripts/lua2dox.lua) which converts these Lua docstrings into something doxygen understands.

We decided to avoid [penlight/ldoc](https://github.com/stevedonovan/LDoc) because (1) it's a big dependency, and (2) it still wouldn't help us convert the docstrings to Vim's `:help` format.

`scripts/gen_vimdoc.py` consumes doxygen XML and generates `:help` files.

Reply:
on current master, Lua docstrings are explaind at:

    :help dev-lua-doc